### PR TITLE
Fix updater GUI

### PR DESCRIPTION
**Pull request by brusherru**
1. Restart button was not working stable: sometimes it does nothing. Now it is fixed.
2. Sometimes we will have to downgrade the Smapp version (e.g. on switching between networks with breaking changes). So it was a bit silly to observe "v0.2.11 Update available: v0.2.10". Now it will tell, that Network supports older version and propose to Downgrade.

### CODE REVIEW

File: app/screens/auth/SwitchNetwork.tsx [hunk 1]
```diff
@@ -132,7 +139,9 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
     />
   ) : (
     <Wrapper>
-      {!!location.state.creatingWallet && <Steps step={Step.SELECT_NETWORK} />}
+      {!!location?.state?.creatingWallet && (
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]:
```suggestion
      {Boolean(location?.state?.creatingWallet) && (
```

File: app/screens/auth/SwitchNetwork.tsx [hunk 2]
```diff
@@ -77,7 +84,7 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
       : [{ label: 'NO NETWORKS AVAILABLE', isDisabled: true }];
 
   const goNext = (genesisID: string | undefined) => {
-    const { creatingWallet, isWalletOnly, mnemonic } = location.state;
+    const { creatingWallet, isWalletOnly, mnemonic } = location?.state || {};
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]: Will be looking much better if we will place this get operation above  , out of function and will replace all places with location?.state?.PROPERTY
```
const { creatingWallet, isWalletOnly, mnemonic } = location?.state || {};
```

File: app/screens/auth/SwitchNetwork.tsx [hunk 3]
```diff
@@ -34,13 +35,21 @@ const BottomPart = styled.div`
...
   const [showLoader, setLoader] = useState(false);
+  const curGenesisId = useSelector(getGenesisID);
+  const curIndex = networks.networks.findIndex(
+    (n) => n.genesisID === curGenesisId
+  );
+  const [selectedItemIndex, setSelectedItemIndex] = useState(
+    curIndex > -1 ? curIndex : 0
+  );
+
+  const { creatingWallet, isWalletOnly, mnemonic } = location?.state || {};
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]:
```suggestion
  const { creatingWallet, isWalletOnly, mnemonic, redirect } = location?.state || {};
```

File: app/screens/auth/SwitchNetwork.tsx [hunk 4]
```diff
@@ -95,14 +103,14 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
     }
     if (genesisID?.length && isWalletOnly) {
       return history.push(AuthPath.ConnectToAPI, {
-        redirect: location?.state?.redirect,
+        redirect: location.state?.redirect,
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]:
```suggestion
        redirect,
```

File: app/screens/auth/SwitchNetwork.tsx [hunk 5]
```diff
@@ -95,14 +103,14 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
...
         genesisID,
         isWalletOnly,
         creatingWallet,
         mnemonic,
       });
     }
-    return history.push(location?.state?.redirect || AuthPath.Unlock);
+    return history.push(location.state?.redirect || AuthPath.Unlock);
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]:
```suggestion
    return history.push(redirect || AuthPath.Unlock);
```

File: app/screens/auth/SwitchNetwork.tsx [hunk 6]
```diff
@@ -132,7 +140,9 @@ const SwitchNetwork = ({ history, location }: AuthRouterParams) => {
     />
   ) : (
     <Wrapper>
-      {!!location.state.creatingWallet && <Steps step={Step.SELECT_NETWORK} />}
+      {Boolean(location.state?.creatingWallet) && (
```
Comment (maparr) [inline app/screens/auth/SwitchNetwork.tsx]:
```suggestion
      {creatingWallet && (
```

**Comment by brusherru** [comment]
To test it you can pick this signed artifacts:
- Windows: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh%20Setup%200.2.11-test-autoupdater-restart.exe
- macOS: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart.dmg
- Linux: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/spacemesh_app_0.2.11-test-autoupdater-restart_amd64.deb

And then just try to update to something else ;)

Also, gh runner-images was updated recently and made deprecated `altool` was not working anymore.
So [the commit 32d5f9d](https://github.com/spacemeshos/smapp/pull/1148/commits/32d5f9d6ce0557c4414befa8ecc8b22521632cc6) fixes it by using `notarytool` instead.

p.s.
The latest code-style tweaks are not in these release builds. But they should not affect anything :)

**Comment by maparr** [comment]
Hi @brusherru 
**question**: got 3 out + 1 inner Quit modal, is it reproducible on your side? Did the restart app after the Download update

Screenshot 2023-02-17 at 23 31 43


**not critical** , Is it possible to move it to modal in case of error or , or resize screen  , not sure is it needed to fix, rare case

Screenshot 2023-02-17 at 23 29 33

**Comment by monikasmolarek** [comment]
I reproduced it, but I believe it should be fixed with #1150....

Screenshot 2023-02-20 at 20 08 28


```
[2023-02-20 20:07:56.956] [info]  / requested
[2023-02-20 20:07:56.962] [info]  /1676920076926-3832.zip requested
[2023-02-20 20:07:56.962] [info]  /1676920076926-3832.zip requested by Squirrel.Mac, pipe /Users/monikasmolarek/Library/Application Support/Caches/spacemesh_app-updater/pending/Spacemesh-0.2.11-mac.zip
[2023-02-20 20:09:08.473] [error] reactions/handleCloseApp, in promptBeforeClose, error: {} 
[2023-02-20 20:09:10.041] [error] reactions/handleCloseApp, in promptBeforeClose, error: {} 
[2023-02-20 20:09:10.084] [info]  Proxy server for native Squirrel.Mac is closed (was started to download https://storage.googleapis.com/smapp/v0.2.11/Spacemesh-0.2.11-mac.zip)
[2023-02-20 20:09:11.143] [error] reactions/handleCloseApp, in promptBeforeClose, error: {} 
```

**Comment by monikasmolarek** [comment]
Can't test on windows because #1140 still open, if I understand correctly 

Screenshot 2023-02-20 at 21 16 59

**Comment by brusherru** [comment]
> Can't test on windows because #1140 still open, if I understand correctly

@monikasmolarek you can try to update to `v0.2.11-fix-checksum` and it should work.
But anyway, I've merged a few PRs, so I'm going to rebase this PR as well and push the tag — then we can try all the fixes together.

For example, this issue with many prompts (#1150) on Restart should disappear

**Comment by brusherru** [comment]
All done for tests / reviews.
Updated binaries are available through the same links:
- Windows: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh%20Setup%200.2.11-test-autoupdater-restart.exe
- macOS: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart.dmg
- Linux: https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/spacemesh_app_0.2.11-test-autoupdater-restart_amd64.deb

**Comment by brusherru** [comment]
### Latest signed test builds:
- Windows: [x64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh%20Setup%200.2.11-test-autoupdater-restart.exe)
- macOS: [x64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart.dmg) [arm64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart-arm64.dmg)
- Linux
   - AppImage [x64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart.AppImage) [arm64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/Spacemesh-0.2.11-test-autoupdater-restart-arm64.AppImage)
   - deb [x64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/spacemesh_app_0.2.11-test-autoupdater-restart_amd64.deb) [arm64](https://storage.googleapis.com/smapp/v0.2.11-test-autoupdater-restart/spacemesh_app_0.2.11-test-autoupdater-restart_arm64.deb)

---
You should be able to update any of them to version `0.2.11-test.appimages` without any issues.

**Review by monikasmolarek — APPROVED** [review]

**Pull request merged**